maxq10xx/wolfssl-lwip-client.c

```diff
@@ -0,0 +1,318 @@
+/* wolfssl-lwip-client.c
+ *
+ * Based on client-tls-pkcallback.c
+ *
+ * Copyright (C) 2006-2024 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This example shows how to write a simple TLS client that uses the features
+ * of the Analog Devices MAXQ1065 and 1080 USS. Note that this is not a
+ * stand-alone application. This is part of an example that can be found in the
+ * SDK supplied by Analog Devices. */
+
+#include "lwip/opt.h"
+
+#if !NO_SYS /* don't build if not configured to run an OS in lwipopts.h */
+
+/***** Includes *****/
+#include <stdio.h>
+
+#include <wolfssl/wolfcrypt/sha256.h>

```

Comment:
Is this linking against a wolfSSL library that was compiled without autoconf? Lets add an include of wolfssl/options.h as first wolfSSL include for autoconf builds, possibly gated by check of WOLFSSL_USER_SETTINGS if a non-autoconf build is expected.

Reply:
Its always going to be built against a library built with ./configure.  That said, you're right....lets future proof.  Stay tuned for next commit.